selftests/unit/test_runner_queue.py

```diff
@@ -28,7 +27,11 @@ def _run_test(self, factory):
         :param factory: the Avocado Test factory
         :return: the last queue message from the test
         """
-        queue = queues.SimpleQueue()
+        if hasattr(multiprocessing, 'get_context'):
+            ctx = multiprocessing.get_context('spawn')
+            queue = multiprocessing.queues.SimpleQueue(ctx=ctx)     # pylint: disable=E1123
+        else:
+            queue = multiprocessing.queues.SimpleQueue()
```

Comment:
Ok, I know this is pretty much the core implementation on `avocado.core.runner`,  but we are forcing a context when the multiprocessing should be in charge of probing it, according to the O.S. On Windows, `spawn` is the default. On Linux, `fork` is the default. To do so, `SimpleQueue` is now (Py3) a top-level function which gets a context by its own:
(https://github.com/python/cpython/blob/3.4/Lib/multiprocessing/context.py)
```
    def SimpleQueue(self):
        '''Returns a queue object'''
        from .queues import SimpleQueue
        return SimpleQueue(ctx=self.get_context())
```
(that function is imported by `multiprocessing/__init__.py`).

And why to use the defaults? According to the docs:

- spawn: Starting a process using this method is rather slow compared to using *fork* or *forkserver*. Available on Unix and Windows. The default on Windows.
- fork: Available on Unix only. The default on Unix.

So we should probably have something like this (also in `avocado.core.runner`):

```python
try:
    queue = multiprocessing.SimpleQueue()
except:
    queue = multiprocessing.queues.SimpleQueue()
```

But maybe you have some reason to force spawn on *nix?

Reply:
@apahim you're right about that.  Thanks for spotting this.   I've sent a PR with that specific change to the `core/runner.py` code:

https://github.com/avocado-framework/avocado/pull/2451